src/Symfony/Component/Console/Application.php

```diff
@@ -271,25 +289,22 @@ public function doRun(InputInterface $input, OutputInterface $output)
                     $e = $event->getError();
                 }
 
-                throw $e;
-            }
-
-            $alternative = $alternatives[0];
-
-            $style = new SymfonyStyle($input, $output);
-            $style->block(sprintf("\nCommand \"%s\" is not defined.\n", $name), null, 'error');
-            if (!$style->confirm(sprintf('Do you want to run "%s" instead? ', $alternative), false)) {
-                if (null !== $this->dispatcher) {
-                    $event = new ConsoleErrorEvent($input, $output, $e);
-                    $this->dispatcher->dispatch($event, ConsoleEvents::ERROR);
-
-                    return $event->getExitCode();
+                try {
+                    if ($e instanceof CommandNotFoundException && $namespace = $this->findNamespace($name)) {
+                        $helper = new DescriptorHelper();
+                        $helper->describe($output, $this, [
```

Comment:
As this part of reporting an error, we should probably send the description to stderr when supported, to avoid having it be piped to another command.